packages/react/src/hooks/buckets.ts

```diff
@@ -100,61 +100,79 @@ export function useQueryObjects(
     enabled?: boolean | (() => boolean);
   },
 ) {
-  const chainId = useChainId();
-  const contractAddress =
-    bucketManagerAddress[chainId as keyof typeof bucketManagerAddress];
-
   const args = [
-    bucket,
     options?.prefix ?? "",
     options?.delimiter ?? "/",
     options?.startKey ?? "",
     BigInt(options?.limit ?? 100),
   ] satisfies QueryObjectsArgs;
 
-  return useReadContract({
-    address: contractAddress,
-    abi: bucketManagerAbi,
+  const { data, ...rest } = useReadContract({
+    address: bucket,
+    abi: iBucketFacadeAbi,
     functionName: "queryObjects",
     args: args,
     query: {
       enabled: options?.enabled,
     },
   });
+
+  return {
+    data: data
+      ? {
+          ...data,
+          objects: data.objects.map((obj) => ({
+            ...obj,
+            state: {
+              ...obj.state,
+              blobHash: hexToBase32(obj.state.blobHash),
```

Comment:
here's where the `fvm` package + base32 decoding comes into play. we have to override the default response from the contract and convert the bytes32 blob has to a base32 encoded string.